tls.md

```diff
@@ -155,21 +213,26 @@ Public/internal API services:
+As mentioned above - certificates, including TLS certificates (`tls.crt`), private keys (`tls.key`), and CA certificates (`ca.crt`), are stored in secrets. These secrets are mounted into the containers of the services that require them. Depending on how the service gets started (using kolla or not), the cert and key get either mounted to a directory and using kolla to put into its final place, or directly mounted to where the service expects it. The CA bundle is always mounted as the global CA bundle to the final destination (bypassing kolla).
 
 For k8s service a virtual host gets configured using the service name with its corresponding certificate.
 
-Like with other config secrets/config maps a hash gets used to identify if the certificate changed and the deployment pod has to be restarted. For this, the service operators index the named secret fields to be able to watch those if they change. If a change happenes which result in a hash change, the default action is to restart the deployment. There might be services which will handle this different.
+Like with other config secrets and config maps, a hash is used to identify if the certificate changed and the deployment pod has to be restarted. For this, the service operators index the named secret fields to be able to watch those if they change. If a change is detected which results in a hash change, the default action is to restart the deployment in order to use the new certificates. There might be services which will handle this differently, but curerntly there's no option to put new certificates in use without restarting the deployment.
 
-If internal tls is enabled, using `spec.tls.endpoint.internal.enabled: true`, routes get configured with `spec.tls.termination: reencrypt` and CA added to `spec.tls.destinationCACertificate` to be able to validate the certificate of the k8s service.
+If internal tls is enabled, using `spec.tls.podlevel.enabled: true`, routes get configured with `spec.tls.termination: reencrypt` and CA added to `spec.tls.destinationCACertificate` to be able to validate the certificate of the k8s service.
 
-#### Deployments not using k8s services
+## Deployments Using k8s Services
+
+The configuration of the service is reflecting the k8s service name, not the individual pod name. With this all pods in a scaling resource will share the same service certificate. 
+
+For each of the k8s services the openstack-operator requests a certificate for the hostname `<service>.<namespace>.svc` using the internal and the public `CertManager` issuer. For each cert request, CertManger will create a secret holding a `tls.crt`, `tls.key` and `ca.crt`. Those secrets, certificate and the combined CA bundle described above, can be passed into the service operators CRD using the tls section.
```

Comment:
we now add both names `<service>.<namespace>.svc` `<service>.<namespace>.svc.cluster.local`